packages/manager/src/views/WebPropertyEnvPage/components/EditAccess/EditMemberAccess.tsx

```diff
@@ -0,0 +1,206 @@
+  editMemberName: string;
+  propertyIdentifier: string;
+  memberList: any;
+}
+type ColumnNames = {
+  NAME: string;
+  APIKEY_CREATION: string;
+  APIKEY_DELETION: string;
+  PERMISSION_CREATION: string;
+  PERMISSION_DELETION: string;
+  ENV_CREATION: string;
+  ENV_SYNC: string;
+  [key: string]: string;
+};
+type GroupItem1 = {
+  email: string;
+  name: string;
+  role: string;
+  PERMISSION_CREATION: boolean;
+  PERMISSION_DELETION: boolean;
+  ENV_SYNC: boolean;
+  ENV_CREATION: boolean;
+  APIKEY_DELETION: boolean;
+  APIKEY_CREATION: boolean;
+};
+export const EditMemberAccess = ({
+  onClose,
+  editMemberName,
+  propertyIdentifier,
+  memberList
+}: Props): JSX.Element => {
+  const [expanded, setExpanded] = useState('def-list-toggle2');
+  const [addAccess, setAddAccess] = useState<string[]>([]);
+  const [deleteAccess, setDeleteAccess] = useState<string[]>([]);
+  const useAddPermission1 = useAddPermission(propertyIdentifier);
+  const deleteMember = useDeleteMember(propertyIdentifier);
+  const data1 = memberList.data.filter((e: GroupItem1) => e.name === editMemberName);
+  const [group, setGroup] = useState({ data: data1 });
+  const columnNames: ColumnNames = {
+    NAME: 'Name',
+    // ROLE: 'Role',
+    APIKEY_CREATION: 'APIKEY_CREATION',
+    APIKEY_DELETION: 'APIKEY_DELETION',
+    PERMISSION_CREATION: 'PERMISSION_CREATION',
+    PERMISSION_DELETION: 'PERMISSION_DELETION',
+    ENV_CREATION: 'ENV_CREATION',
+    ENV_SYNC: 'ENV_SYNC'
+  };
+  const onToggleAccordian = async (id: string) => {
+    if (id === expanded) {
+      setExpanded('');
+    } else {
+      setExpanded(id);
+    }
+  };
+  const handleChange = (
+    checked: boolean,
+    event: React.FormEvent<HTMLInputElement>,
+    email: string
+  ) => {
+    const target = event.currentTarget as HTMLInputElement;
+    const temp = group.data;
+    const objIndex = temp.findIndex((e: GroupItem1) => e.email === email);
+    temp[objIndex][target.name] = checked;
+    setGroup({ data: temp });
+    if (checked) {
+      setAddAccess([...addAccess, target.name]);
+      setDeleteAccess(deleteAccess.filter((value) => value !== target.name));
+    } else {
+      setDeleteAccess([...deleteAccess, target.name]);
+      setAddAccess(addAccess.filter((value) => value !== target.name));
+    }
+  };
+  interface GroupData {
+    email: string;
+    name: string;
+    role: string;
+    [key: string]: unknown;
+  }
+  const handleSubmit = () => {
+    const addPerm = group.data.map(({ email, name, role, ...rand }: GroupData) => ({
+      email,
+      name,
+      actions: Object.keys(rand)
+    }));
+    const addData = { propertyIdentifier, permissionDetails: addPerm };
+    const deletePerm = group.data.map(({ email, name, role, ...rand }: GroupData) => ({
+      email,
+      name,
+      actions: Object.keys(rand).filter((value) => rand[value] === false)
+    }));
+    const deleteData = { propertyIdentifier, permissionDetails: deletePerm };
+    try {
```

Comment:
This try catch is not needed anymore as this became a .then based syntax. U can remove it.

Reply:
done